crates/primitives-traits/src/signature.rs

```diff
@@ -0,0 +1,43 @@
+//! Signature abstraction
+
+use alloy_primitives::{Address, Parity, B256, U256};
+
+/// Reth extension for alloy type [`Signature`](alloy_primitives::Signature).
+pub trait Signature: Sized + Send + Sync {
```

Comment:
will this be useful once we transitioned to 

https://github.com/alloy-rs/core/issues/788 

@klkvr ?

Reply:
still need to get the op logic out of this file somehow https://github.com/paradigmxyz/reth/blob/main/crates/primitives/src/transaction/signature.rs

Reply:
I'm wondering if op-related logic belongs to transaction types? i.e OpSignedTransaction knows how to encode itself with signature, so we can just put all op-specific logic to OpTx::decode/OpTx::recover_signer and still use usual ECDSA signature?

the idea behind https://github.com/alloy-rs/core/issues/788 is that we don't need logic related to legacy parity on core signature type as it can be handled by `TxLegacy` itself

Reply:
alright, adding the functions from the [file](https://github.com/paradigmxyz/reth/blob/main/crates/primitives/src/transaction/signature.rs) to the `SignedTransaction` trait would be ok